src/ReinforcementLearningCore/src/policies/agent_srt_cache.jl

```diff
@@ -0,0 +1,38 @@
+mutable struct SRT{S,R,T}
```

Comment:
I can't help but wonder if using Union types is not detrimental to performance ? Sure it's better the the former NamedTuple of Anys. But I was wondering if there would not be a way to initialize this trace to undef with a boolean field that asserts whether the cache still has undef variables. Thus, we wouldn't need to check for isnothing at all (simply set the field to true at the PostActStage).

Reply:
This isn't a union type, it's a type parameter for each of `S`, `R` and `T`, but each is type-pure.